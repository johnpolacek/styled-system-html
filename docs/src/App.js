import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from './Theme'
import Header from './components/Header';
import Footer from './components/Footer';
import Demo from './components/Demo';
import { Flex, Box } from 'grid-styled'
import { Div, Main, H3, P, Pre, Code, A } from 'styled-system-html'

class App extends Component {
  render() {
    return (
      <div className="App">
        <ThemeProvider theme={theme}>
          <Div>
            <Header />
            <Main mx="auto" p={4} maxWidth='1200px'>
              <H3 pb={3}>Usage</H3>
              <P pb={2} width={[1,1,1/2]}><Code p={3} bg="gray0" border='1px solid' borderColor="gray1">npm i styled-system-html</Code></P>
              <Flex flexWrap="wrap" pb={[3,3,3,4]}>
                <Box width={[1,1,1,1/2]} pr={[0,0,0,4]}>
                  <P>Create a <A color="blue" href="https://www.styled-components.com/docs/advanced">theme</A> with <A color="blue" href="http://jxnblk.com/styled-system/api">settings</A> for colors, typography and spacing.</P>
                  <Pre p={3} bg="gray0" border='1px solid' borderColor="gray1"><Code>{`const theme = {
  colors : {'primary':'#071'},
  space : [0,4,8,16,32,64,128]
}`}</Code></Pre>
                </Box>
                <Box width={[1,1,1,1/2]} pl={[0,0,0,4]}>
                  <P>Apply your theme to the element components.</P>
                  <Pre mb={4} p={3} bg="gray0" border='1px solid' borderColor="gray1"><Code>{`<H2 pb={2} color="primary">Hello</H2>`}</Code></Pre>
                  <P>For more info, see <A color="blue" fontWeight="bold" href="http://jxnblk.com/styled-system/">Styled System</A> by <A color="blue" fontWeight="bold" href="http://jxnblk.com">Brent Jackson</A>.</P>
                </Box>
              </Flex>
              <H3 pb={3} m={0}>Live Code Example</H3>
              <Demo />
            </Main>
            <Footer />
          </Div>
        </ThemeProvider>
      </div>
    )
  }
}

export default App
