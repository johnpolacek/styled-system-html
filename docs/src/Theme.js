// Typography
const fonts = {
  sans:`-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, sans-serif`,
  serif:`Garamond, Times New Roman, serif`,
}

export const font = fonts.sans;
export const fontFamilies = fonts;
export const monospace = `"SF Mono", "Roboto Mono", Menlo, monospace`

export const fontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72,
  96
]

export const weights = [
  400,
  700
]

// Colors
export const colors = {
  "base":"#0077cc",
  "blue":"#0077cc",
  "blue0":"#e4f1fa",
  "blue1":"#c7e1f4",
  "blue2":"#a5cfed",
  "blue3":"#7dbae5",
  "blue4":"#4b9fdb",
  "blue5":"#0077cc",
  "blue6":"#006bb8",
  "blue7":"#005ea1",
  "blue8":"#004d84",
  "blue9":"#00375e",
  "black":"#000",
  "cyan":"#00ddcc",
  "cyan0":"#edfdfb",
  "cyan1":"#c5f7f3",
  "cyan2":"#94f1ea",
  "cyan3":"#55e8dd",
  "cyan4":"#00dbca",
  "cyan5":"#00c8b9",
  "cyan6":"#00b3a5",
  "cyan7":"#009a8e",
  "cyan8":"#007a70",
  "cyan9":"#004842",
  "gray":"#adb4b9",
  "gray0":"#f9f9fa",
  "gray1":"#eceeef",
  "gray2":"#dee1e3",
  "gray3":"#cfd3d7",
  "gray4":"#bfc5c9",
  "gray5":"#adb4b9",
  "gray6":"#98a1a8",
  "gray7":"#7f8b93",
  "gray8":"#606e79",
  "gray9":"#374047",
  "green":"#11cc00",
  "green0":"#e6fae4",
  "green1":"#caf4c6",
  "green2":"#aaeda4",
  "green3":"#84e57b",
  "green4":"#54da48",
  "green5":"#11cc00",
  "green6":"#0fb900",
  "green7":"#0ea200",
  "green8":"#0b8700",
  "green9":"#086100",
  "orange":"#ff9900",
  "orange0":"#fff9ef",
  "orange1":"#ffebcd",
  "orange2":"#ffdca6",
  "orange3":"#ffca7b",
  "orange4":"#ffb647",
  "orange5":"#ff9c07",
  "orange6":"#e68a00",
  "orange7":"#c57600",
  "orange8":"#9c5e00",
  "orange9":"#5c3700",
  "pink":"#cc0077",
  "pink0":"#fbeaf4",
  "pink1":"#f6d2e7",
  "pink2":"#f0b6d8",
  "pink3":"#ea94c6",
  "pink4":"#e066ad",
  "pink5":"#cc0077",
  "pink6":"#b9006c",
  "pink7":"#a2005f",
  "pink8":"#87004e",
  "pink9":"#610039",
  "red":"#dd0000",
  "red0":"#fbeaec",
  "red1":"#f6d3d6",
  "red2":"#f1b8bc",
  "red3":"#ea969d",
  "red4":"#e16973",
  "red5":"#cc0011",
  "red6":"#b9000f",
  "red7":"#a2000e",
  "red8":"#87000b",
  "red9":"#610008",
  "violet":"#5500cc",
  "violet0":"#f1eafb",
  "violet1":"#e1d2f6",
  "violet2":"#cfb6f0",
  "violet3":"#b895ea",
  "violet4":"#9a67e1",
  "violet5":"#5500cc",
  "violet6":"#4d00b8",
  "violet7":"#4300a1",
  "violet8":"#370085",
  "violet9":"#27005f",
  "white":"#fff",
  "yellow":"#ffee33",
  "yellow0":"#fffbd1",
  "yellow1":"#fff159",
  "yellow2":"#f4e431",
  "yellow3":"#e5d62e",
  "yellow4":"#d5c72b",
  "yellow5":"#c3b627",
  "yellow6":"#aea223",
  "yellow7":"#958b1e",
  "yellow8":"#766e18",
  "yellow9":"#45400e"
}

export const breakpoints = [
  '480px',
  '600px',
  '720px',
  '840px',
  '960px',
  '1080px',
  '1280px'
]

export const space = [
  0,
  4,
  8,
  16,
  32,
  64,
  128,
]

export const radius = 4

export default {
  font,
  monospace,
  fontFamilies,
  weights,
  fontSizes,
  colors,
  breakpoints,
  space,
  radius
}
