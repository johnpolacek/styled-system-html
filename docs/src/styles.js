import { css } from 'styled-components'
import { style, responsiveStyle, space, width, fontSize, color, fontWeight, textAlign, borderColor, borderWidth, boxShadow, hover, focus, active, disabled } from 'styled-system'

export const height = responsiveStyle({ prop:'height',cssProperty:'height' })
export const maxWidth = responsiveStyle({ prop: 'mw', cssProperty: 'maxWidth' })
export const fontStyle = style({ prop: 'fontStyle', cssProperty: 'fontStyle' })
export const overflow = responsiveStyle({ prop: 'overflow', cssProperty: 'overflow' })

export const wrapperStyles = css` ${space} ${width} ${height} ${fontStyle} ${fontSize} ${textAlign} ${color} ${maxWidth} ${height} ${boxShadow} ${borderColor} ${borderWidth} ${overflow} `
export const textStyles = css` ${space} ${width} ${fontWeight} ${fontStyle} ${fontSize} ${textAlign} ${color} ${boxShadow} ${borderColor} ${borderWidth} `
export const textStylesPseudo = css` ${textStyles} ${hover} ${focus} ${active} ${disabled} `
export const buttonStyles = css`
  display: inline-flex;
  text-decoration: none;
  margin: 0;
  padding: 8px 16px;
  color: white;
  border-radius: 6px;
  cursor: pointer;
  font-size:16px;
  border:none;
  ${textStylesPseudo}
`