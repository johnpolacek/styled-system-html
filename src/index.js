import React from 'react'
import tags from 'html-tags'
import system from 'system-components'

const El = system(
	'space',
	'width',
	'fontSize',
	'textColor',
	'bgColor',
	'color',
	// typography
	'fontFamily',
	'textAlign',
	'lineHeight',
	'fontWeight',
	'letterSpacing',
	// layout
	'display',
	'maxWidth',
	'minWidth',
	'height',
	'maxHeight',
	'minHeight',
	'size',
	'ratio',
	'verticalAlign',
	// flexbox
	'alignItems',
	'alignContent',
	'justifyContent',
	'flexWrap',
	'flexDirection',
	'flex',
	'flexBasis',
	'justifySelf',
	'alignSelf',
	'order',
	// grid
	'gridGap',
	'gridColumnGap',
	'gridRowGap',
	'gridColumn',
	'gridRow',
	'gridAutoFlow',
	'gridAutoColumns',
	'gridAutoRows',
	'gridTemplateColumns',
	'gridTemplateRows',
	// borders
	'border',
	'borderTop',
	'borderRight',
	'borderBottom',
	'borderLeft',
	'borders',
	'borderColor',
	'borderRadius',
	// misc
	'boxShadow',
	'opacity',
	'background',
	'backgroundImage',
	'backgroundSize',
	'backgroundPosition',
	'backgroundRepeat',
	// position
	'position',
	'zIndex',
	'top',
	'right',
	'bottom',
	'left',
	// variants
	'textStyle',
	'colorStyle',
	'buttonStyle'
)

tags.forEach(tag => {
    module.exports[tag.charAt(0).toUpperCase() + tag.slice(1)] = props => <El is={tag} {...props} />
})

module.exports['Text'] = props => <El is={'p'} {...props} />
module.exports['UL'] = props => <El is={'ul'} {...props} />
module.exports['OL'] = props => <El is={'ol'} {...props} />
module.exports['LI'] = props => <El is={'li'} {...props} />
module.exports['DL'] = props => <El is={'dl'} {...props} />
module.exports['DT'] = props => <El is={'dt'} {...props} />
module.exports['THead'] = props => <El is={'thead'} {...props} />
module.exports['TBody'] = props => <El is={'tbody'} {...props} />
module.exports['TH'] = props => <El is={'th'} {...props} />
module.exports['TR'] = props => <El is={'tr'} {...props} />
module.exports['TD'] = props => <El is={'td'} {...props} />
module.exports['TextArea'] = props => <El is={'textarea'} {...props} />
module.exports['FigCaption'] = props => <El is={'figcaption'} {...props} />