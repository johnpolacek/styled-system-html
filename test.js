import React from 'react';
import ReactDOM from 'react-dom';
import { create as render } from 'react-test-renderer'
import tags from 'html-tags'

const renderJSON = el => render(el).toJSON()

import * as elements from './src'

tags.forEach(tag => (
	it('should render '+tag, ()  => {
		const json = renderJSON(
	    	React.createElement(elements[tag.charAt(0).toUpperCase() + tag.slice(1)])
		)
		expect(json).toMatchSnapshot()
	})
))

it('should render Text', () => {
	const json = renderJSON(
			React.createElement(elements['Text'])
		)
	expect(json).toMatchSnapshot()
})

it('should render Image', () => {
	const json = renderJSON(
			React.createElement(elements['Image'])
		)
	expect(json).toMatchSnapshot()
})

it('should render UL', () => {
	const json = renderJSON(
		React.createElement(elements['UL'])
	)
	expect(json).toMatchSnapshot()
})

it('should render OL', () => {
	const json = renderJSON(
		React.createElement(elements['OL'])
	)
	expect(json).toMatchSnapshot()
})

it('should render LI', () => {
	const json = renderJSON(
		React.createElement(elements['LI'])
	)
	expect(json).toMatchSnapshot()
})

it('should render DL', () => {
	const json = renderJSON(
		React.createElement(elements['DL'])
	)
	expect(json).toMatchSnapshot()
})

it('should render DT', () => {
	const json = renderJSON(
		React.createElement(elements['DT'])
	)
	expect(json).toMatchSnapshot()
})

it('should render THead', () => {
	const json = renderJSON(
		React.createElement(elements['THead'])
	)
	expect(json).toMatchSnapshot()
})

it('should render TBody', () => {
	const json = renderJSON(
		React.createElement(elements['TBody'])
	)
	expect(json).toMatchSnapshot()
})

it('should render TH', () => {
	const json = renderJSON(
		React.createElement(elements['TH'])
	)
	expect(json).toMatchSnapshot()
})

it('should render TR', () => {
	const json = renderJSON(
		React.createElement(elements['TR'])
	)
	expect(json).toMatchSnapshot()
})

it('should render TD', () => {
	const json = renderJSON(
		React.createElement(elements['TD'])
	)
	expect(json).toMatchSnapshot()
})

it('should render TextArea', () => {
	const json = renderJSON(
		React.createElement(elements['TextArea'])
	)
	expect(json).toMatchSnapshot()
})

it('should render FigCaption', () => {
	const json = renderJSON(
		React.createElement(elements['FigCaption'])
	)
	expect(json).toMatchSnapshot()
})



