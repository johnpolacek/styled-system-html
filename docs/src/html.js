import styled from 'styled-components'
import { fontSize, color, textAlign, borderColor, borderWidth, boxShadow } from 'styled-system'
import { Flex as _Flex, Box as _Box } from 'grid-styled'
import { height, maxWidth, overflow, wrapperStyles, textStyles, textStylesPseudo, buttonStyles } from './styles'

export const Flex = styled(_Flex)([], fontSize, color, maxWidth, boxShadow, borderColor, borderWidth)
export const Box = styled(_Box)([], fontSize, color, textAlign, maxWidth, height, boxShadow, borderColor, borderWidth, overflow)
export const Div = styled(_Box)([], fontSize, color, textAlign, maxWidth, height, boxShadow, borderColor, borderWidth, overflow)
export const Header = styled.header` ${wrapperStyles} `
export const Main = styled.main` ${wrapperStyles} `
export const Footer = styled.footer` ${wrapperStyles} `

export const Section = styled.section` ${wrapperStyles} `
export const Article = styled.article` ${wrapperStyles} `
export const Aside = styled.aside` ${wrapperStyles} `
export const Blockquote = styled.blockquote` margin:0; ${wrapperStyles} `
export const HGroup = styled.hgroup` margin:0; ${wrapperStyles} `
export const H1 = styled.h1` ${textStyles} `
export const H2 = styled.h2` ${textStyles} `
export const H3 = styled.h3` ${textStyles} `
export const H4 = styled.h4` ${textStyles} `
export const H5 = styled.h5` ${textStyles} `
export const H6 = styled.h6` ${textStyles} `
export const Text = styled.p` ${textStyles} `
export const P = styled.p` padding-bottom:16px; ${textStyles} `
export const A = styled.a` ${textStylesPseudo}`
export const Span = styled.span` ${textStyles} `
export const Em = styled.em` ${textStyles} `
export const Strong = styled.strong` ${textStyles} `
export const Small = styled.small` ${textStyles} `
export const Sub = styled.sub` ${textStyles} `
export const Sup = styled.sup` ${textStyles} `
export const Underline = styled.u` ${textStyles} `
export const Strike = styled.s` ${textStyles} `
export const Mark = styled.mark` ${textStyles} `
export const Abbr = styled.abbr` ${textStyles} `
export const Quote = styled.q` ${textStyles} `
export const Cite = styled.cite` ${textStyles} `
export const Definition = styled.dfn` ${textStyles} `
export const Data = styled.data` ${textStyles} `
export const Time = styled.time` ${textStyles} `
export const Pre = styled.pre` ${textStyles} white-space:pre;-webkit-hyphens:none;-ms-hyphens:none;hyphens:none;word-wrap:normal;word-break:normal;word-spacing:normal;`
export const Code = styled.code` ${textStyles} `
export const Figure = styled.figure`margin:0 ${wrapperStyles} `
export const FigCaption = styled.figcaption` ${textStyles} `
export const UL = styled.ul` ${wrapperStyles} `
export const OL = styled.ol` ${textStyles} `
export const LI = styled.li` padding-bottom:16px; ${textStyles} `
export const DL = styled.dl` ${wrapperStyles} `
export const DT = styled.dt` ${textStyles} `
export const DD = styled.dd` ${textStyles} `

export const Nav = styled.nav` ${wrapperStyles} `
export const Button = styled.button` ${buttonStyles} `
export const LinkButton = styled.a` ${buttonStyles} `
export const InputSubmit = styled.input` ${buttonStyles} `

export const Form = styled.form` ${wrapperStyles} `
export const Fieldset = styled.fieldset` ${wrapperStyles} `
export const Legend = styled.legend` ${textStyles} `
export const Label = styled.label` ${textStyles} `
export const Input = styled.input` ${wrapperStyles} `
export const Select = styled.select` ${wrapperStyles} `
export const Option = styled.option
export const TextArea = styled.textarea` ${wrapperStyles} `

export const Table = styled.table` ${wrapperStyles} `
export const TBody = styled.tbody` ${wrapperStyles} `
export const THead = styled.thead` ${wrapperStyles} `
export const TH = styled.th` ${wrapperStyles} `
export const TR = styled.tr` ${wrapperStyles} `
export const TD = styled.td` ${wrapperStyles} `
export const TFoot = styled.tfoot` ${wrapperStyles} `
export const Colgroup = styled.colgroup` ${wrapperStyles} `
export const Col = styled.col` ${wrapperStyles} `
export const Caption = styled.caption` ${wrapperStyles} `

export const Img = styled.img` ${wrapperStyles} `
export const Image = styled.img` ${wrapperStyles} `
export const Audio = styled.audio` ${wrapperStyles} `
export const Video = styled.video` ${wrapperStyles} `
