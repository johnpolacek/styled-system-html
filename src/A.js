import styled from 'styled-components'
import { textStylesPseudo } from './styles'

const A = styled.a` ${textStylesPseudo} `
export default A