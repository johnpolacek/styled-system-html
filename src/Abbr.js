import styled from 'styled-components'
import { textStyles } from './styles'

const Abbr = styled.abbr` ${textStyles} `
export default Abbr