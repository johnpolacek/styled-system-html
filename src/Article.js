import styled from 'styled-components'
import { wrapperStyles } from './styles'

const Article = styled.article` ${wrapperStyles} `
export default Article