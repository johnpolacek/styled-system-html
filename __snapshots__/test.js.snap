// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DL 1`] = `
<dl
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render DT 1`] = `
<dt
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render FigCaption 1`] = `
<figcaption
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render Image 1`] = `
<img
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render LI 1`] = `
<li
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render OL 1`] = `
<ol
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render TBody 1`] = `
<tbody
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render TD 1`] = `
<td
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render TH 1`] = `
<th
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render THead 1`] = `
<thead
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render TR 1`] = `
<tr
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render Text 1`] = `
<p
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render TextArea 1`] = `
<textarea
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render UL 1`] = `
<ul
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render a 1`] = `
<a
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render abbr 1`] = `
<abbr
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render address 1`] = `
<address
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render area 1`] = `
<area
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render article 1`] = `
<article
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render aside 1`] = `
<aside
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render audio 1`] = `
<audio
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render b 1`] = `
<b
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render base 1`] = `
<base
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render bdi 1`] = `
<bdi
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render bdo 1`] = `
<bdo
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render blockquote 1`] = `
<blockquote
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render body 1`] = `
<body
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render br 1`] = `
<br
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render button 1`] = `
<button
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render canvas 1`] = `
<canvas
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render caption 1`] = `
<caption
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render cite 1`] = `
<cite
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render code 1`] = `
<code
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render col 1`] = `
<col
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render colgroup 1`] = `
<colgroup
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render data 1`] = `
<data
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render datalist 1`] = `
<datalist
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render dd 1`] = `
<dd
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render del 1`] = `
<del
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render details 1`] = `
<details
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render dfn 1`] = `
<dfn
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render dialog 1`] = `
<dialog
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render div 1`] = `
<div
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render dl 1`] = `
<dl
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render dt 1`] = `
<dt
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render em 1`] = `
<em
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render embed 1`] = `
<embed
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render fieldset 1`] = `
<fieldset
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render figcaption 1`] = `
<figcaption
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render figure 1`] = `
<figure
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render footer 1`] = `
<footer
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render form 1`] = `
<form
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h1 1`] = `
<h1
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h2 1`] = `
<h2
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h3 1`] = `
<h3
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h4 1`] = `
<h4
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h5 1`] = `
<h5
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render h6 1`] = `
<h6
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render head 1`] = `
<head
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render header 1`] = `
<header
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render hgroup 1`] = `
<hgroup
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render hr 1`] = `
<hr
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render html 1`] = `
<html
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render i 1`] = `
<i
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render iframe 1`] = `
<iframe
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render img 1`] = `
<img
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render input 1`] = `
<input
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render ins 1`] = `
<ins
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render kbd 1`] = `
<kbd
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render keygen 1`] = `
<keygen
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render label 1`] = `
<label
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render legend 1`] = `
<legend
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render li 1`] = `
<li
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render link 1`] = `
<link
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render main 1`] = `
<main
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render map 1`] = `
<map
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render mark 1`] = `
<mark
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render math 1`] = `
<math
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render menu 1`] = `
<menu
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render menuitem 1`] = `
<menuitem
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render meta 1`] = `
<meta
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render meter 1`] = `
<meter
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render nav 1`] = `
<nav
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render noscript 1`] = `
<noscript
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render object 1`] = `
<object
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render ol 1`] = `
<ol
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render optgroup 1`] = `
<optgroup
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render option 1`] = `
<option
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render output 1`] = `
<output
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render p 1`] = `
<p
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render param 1`] = `
<param
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render picture 1`] = `
<picture
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render pre 1`] = `
<pre
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render progress 1`] = `
<progress
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render q 1`] = `
<q
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render rb 1`] = `
<rb
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render rp 1`] = `
<rp
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render rt 1`] = `
<rt
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render rtc 1`] = `
<rtc
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render ruby 1`] = `
<ruby
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render s 1`] = `
<s
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render samp 1`] = `
<samp
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render script 1`] = `
<script
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render section 1`] = `
<section
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render select 1`] = `
<select
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render slot 1`] = `
<slot
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render small 1`] = `
<small
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render source 1`] = `
<source
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render span 1`] = `
<span
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render strong 1`] = `
<strong
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render style 1`] = `
<style
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render sub 1`] = `
<sub
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render summary 1`] = `
<summary
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render sup 1`] = `
<sup
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render svg 1`] = `
<svg
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render table 1`] = `
<table
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render tbody 1`] = `
<tbody
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render td 1`] = `
<td
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render template 1`] = `
<template
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render textarea 1`] = `
<textarea
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render tfoot 1`] = `
<tfoot
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render th 1`] = `
<th
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render thead 1`] = `
<thead
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render time 1`] = `
<time
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render title 1`] = `
<title
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render tr 1`] = `
<tr
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render track 1`] = `
<track
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render u 1`] = `
<u
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render ul 1`] = `
<ul
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render var 1`] = `
<var
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render video 1`] = `
<video
  className="sc-bdVaJa bDWFJH"
/>
`;

exports[`should render wbr 1`] = `
<wbr
  className="sc-bdVaJa bDWFJH"
/>
`;
